addons/mail/static/src/components/file_uploader/file_uploader_tests.js

```diff
@@ -87,6 +89,48 @@ QUnit.test('no conflicts between file uploaders', async function (assert) {
     );
 });
 
+QUnit.test('does not crash when an attachment is removed before its upload starts', async function (assert) {
+    assert.expect(1);
+
+    const attachmentUploadedPromise = makeTestPromise();
+    await this.start({
+        async mockFetch(resource, init) {
+            const _super = this._super.bind(this, ...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await attachmentUploadedPromise;
+            }
+            return _super();
+        },
+    });
+    const fileUploader = await this.createFileUploaderComponent();
+    const file1 = await createFile({
+        name: 'text1.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    const file2 = await createFile({
+        name: 'text2.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    inputFiles(
+        fileUploader.el.querySelector('.o_FileUploader_input'),
+        [file1, file2]
+    );
+    await nextAnimationFrame();
+    const attachment = this.env.models['mail.attachment'].find(attachment =>
+        attachment.isTemporary &&
+        attachment.filename === 'text2.txt'
+    );
+    await attachment.remove();
+    attachmentUploadedPromise.resolve();
+    assert.strictEqual(
+        this.env.models['mail.attachment'].all().length,
+        1,
+        'Uploaded file should be the only attachment created'
+    );
```

Comment:
Nah, avoid assertion using model API. This is component test, rely on UI. The functionally meaningful assertion of test is that at end of small flow, just the 1st attachment is uploaded.

Crash is a side-effect, an implementation detail. Test will fail if so, which is good enough as is.